src/Automation/Automation/Cmdlet/UpdateManagement/NewAzureAutomationSoftwareUpdateConfiguration.cs

```diff
@@ -40,15 +42,39 @@ public class NewAzureAutomationSoftwareUpdateConfiguration : AzureAutomationBase
         [Parameter(ParameterSetName = AutomationCmdletParameterSets.Linux, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Indicates that the software update configuration targeting Linux operating system machines.")]
         public SwitchParameter Linux { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Indicates that the software update configuration will Only Reboot the machines.")]
+        public SwitchParameter RebootOnly { get; set; }
+
         [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Resource Ids for azure virtual machines.")]
         public string[] AzureVMResourceId { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Pre task.")]
+        public string PreTaskRunbookName { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Post task.")]
+        public string PostTaskRunbookName { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Pre task parameter.")]
+        public Hashtable PreTaskRunbookParameters { get; set; }
```

Comment:
@jemex another issue you'll see in the `SignatureIssues.csv` file is about [noun singularity in parameter names](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/azure-powershell-design-guidelines.md#singularity) -- from that rule:

_Avoid using plural names for parameters whose value is a single element. This includes parameters that take arrays or lists because the user might supply an array or list with only one element._

_Plural parameter names should be used only in those cases where the value of the parameter is always a multiple-element value. In these cases, the cmdlet should verify that multiple elements are supplied, and the cmdlet should display a warning to the user if multiple elements are not supplied._

Reply:
@cormacpayne Thank you so much for the explanation. It make sense. I changed it to singular